libs/ngxtension/linked-query-param/src/linked-query-param.ts

```diff
@@ -0,0 +1,414 @@
+ *
+ * @param key The name of the query parameter.
+ * @param options Configuration options for the signal.
+ * @returns A signal that is linked to the query parameter.
+ */
+export function linkedQueryParam<T = string>(
+	key: string,
+	options: LinkedQueryParamOptions & {
+		parse: ParseFn<T>;
+		serialize: SerializeFn<T>;
+	},
+): WritableSignal<T> & {
+	set: SignalSetFn<T | null>;
+	update: SignalUpdateFn<T | null>;
+};
+
+/**
+ * You cannot use both `defaultValue` and `parse` at the same time.
+ * You should use `parse` instead to handle the default value.
+ *
+ * For example, you cannot do this:
+ *
+ * ```ts
+ * linkedQueryParam('param', { defaultValue: 1, parse: (x) => x ? parseInt(x, 10) : x });
+ * ```
+ *
+ * Instead, you should do this:
+ *
+ * ```ts
+ * linkedQueryParam('param', { parse: (x) => x ? parseInt(x, 10) : 1 });
+ * ```
+ */
+export function linkedQueryParam<T = string>(
+	key: string,
+	options: LinkedQueryParamOptions & {
+		defaultValue: Exclude<T, undefined>;
+		parse: ParseFn<T>;
+		serialize?: SerializeFn<T>;
+	},
+): never;
+
+export function linkedQueryParam<T = string>(
+	key: string,
+	options: LinkedQueryParamOptions & {
+		defaultValue: T;
+		serialize: SerializeFn<T>;
+	},
+): WritableSignal<T | null>;
+
+export function linkedQueryParam<T>(
+	key: string,
+	options: LinkedQueryParamOptions & { defaultValue: T },
+): WritableSignal<T> & {
+	set: SignalSetFn<T | null>;
+	update: SignalUpdateFn<T | null>;
+};
+
+export function linkedQueryParam<T>(
+	key: string,
+	options: LinkedQueryParamOptions & { defaultValue: T | null },
+): WritableSignal<T | null>;
+
+export function linkedQueryParam<T>(
+	key: string,
+	options: LinkedQueryParamOptions & { defaultValue: T | undefined },
+): WritableSignal<T | undefined>;
+
+export function linkedQueryParam<T = string>(
+	key: string,
+	options: LinkedQueryParamOptions & { defaultValue: undefined },
+): WritableSignal<T | null>;
+
+export function linkedQueryParam<T>(
+	key: string,
+	options: LinkedQueryParamOptions & { parse: ParseFn<T> },
+): WritableSignal<T> & {
+	set: SignalSetFn<T | null>;
+	update: SignalUpdateFn<T | null>;
+};
+
+export function linkedQueryParam<T = string>(
+	key: string,
+	options: LinkedQueryParamOptions & { serialize: SerializeFn<T> },
+): WritableSignal<T | null>;
+
+export function linkedQueryParam<T = string>(
+	key: string,
+	options: LinkedQueryParamOptions,
+): WritableSignal<T | null>;
+
+export function linkedQueryParam<T = string>(
+	key: string,
+): WritableSignal<T | null>;
+
+export function linkedQueryParam<T>(
+	key: string,
+	options?: LinkedQueryParamOptions & {
+		defaultValue?: T;
+		parse?: ParseFn<T>;
+		serialize?: SerializeFn<T>;
+	},
+): WritableSignal<T> {
+	const injector = assertInjector(linkedQueryParam, options?.injector);
+
+	if (options?.defaultValue !== undefined && options?.parse) {
+		throw new Error(
+			'linkedQueryParam: You cannot have both defaultValue and parse at the same time!',
+		);
+	}
```

Comment:
nit: I would move this error case to the top of the function even before asserting the Injector